syntax = "proto2";

package fast_semantic_segmentation.protos;

message InputReader {
    optional bool shuffle = 1 [default=true];
    optional uint32 queue_capacity = 2 [default=2000];
    optional uint32 min_after_dequeue = 3 [default=1000];
    optional uint32 num_epochs = 4 [default=0];
    optional uint32 num_readers = 5 [default=1];
    optional uint32 num_parallel_calls = 6 [default=1];
    optional uint32 shuffle_buffer = 7 [default=2048];
    optional TFRecordInputReader tf_record_input_reader = 8;
    optional uint32 num_examples = 9;
}

message TFRecordInputReader {
    repeated string input_path = 1;
}